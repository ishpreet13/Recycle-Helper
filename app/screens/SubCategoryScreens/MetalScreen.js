import React, { useEffect } from "react";
import { View, StyleSheet, Text } from "react-native";
import { BackHandler } from "react-native";

import AppButton from "../../components/AppButton";
import Screen from "../../components/Screen";
import Styles from "../../config/Styles";
import colors from "../../config/colors";

function MetalScreen({ navigation }) {
  const handleBackButtonClick = () => {
    navigation.navigate("MainCategoryList");
    return true;
  };
  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", handleBackButtonClick);
    return () => {
      BackHandler.removeEventListener(
        "hardwareBackPress",
        handleBackButtonClick
      );
    };
  }, []);
  return (
    <Screen>
      <View style={styles.container}>
        <View style={styles.heading}>
          <Text style={Styles.heading}>
            Select Metal Category for Recycle Instructions
          </Text>
        </View>
        <View style={styles.buttonContainer}>
          <AppButton
            title="Aerosol Cans"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Aerosol Cans",
              })
            }
          ></AppButton>
          <AppButton
            title="Aluminum Cans"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Aluminum Cans",
              })
            }
          ></AppButton>
          <AppButton
            title="Aluminum Foil"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Aluminum Foil",
              })
            }
          ></AppButton>
          <AppButton
            title="Aluminium Takeout Containers"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Aluminium Takeout Containers",
              })
            }
          ></AppButton>
          <AppButton
            title="Metal Caps and Lids"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Metal Caps and Lids",
              })
            }
          ></AppButton>
          <AppButton
            title="Scrap Metal"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Scrap Metal",
              })
            }
          ></AppButton>
          <AppButton
            title="Tin or Steel Cans"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Tin or Steel Cans",
              })
            }
          ></AppButton>
        </View>
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    width: "90%",
  },
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: colors.white,
  },
  heading: {
    marginTop: 20,
    marginBottom: 20,
    paddingLeft: 2,
    paddingRight: 2,
  },
});
export default MetalScreen;
