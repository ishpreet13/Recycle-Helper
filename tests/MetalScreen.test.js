import "react-native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import MetalScreen from "../app/screens/SubCategoryScreens/MetalScreen";

act(() => {
  describe("Testing MetalScreen ", () => {
    let wrapper;
    it("should render", async () => {
      act(() => {
        wrapper = renderer.create(<MetalScreen />);
      });
      expect(wrapper.toJSON()).toBeTruthy();
    });
  });
});

act(() => {
  it("MetalScreen should render correctly", async () => {
    let tree;
    act(() => {
      tree = renderer.create(<MetalScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Aerosol Cans"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Aerosol Cans",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Aluminum Cans"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Aluminum Cans",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Aluminum Foil"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Aluminum Foil",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Aluminium Takeout Containers"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Aluminium Takeout Containers",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Metal Caps and Lids"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Metal Caps and Lids",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Scrap Metal"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Scrap Metal",
    });
  });
});

describe("MetalScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<MetalScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Tin or Steel Cans"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Tin or Steel Cans",
    });
  });
});
