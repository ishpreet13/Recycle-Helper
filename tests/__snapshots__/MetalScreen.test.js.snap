// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetalScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 2,
          "paddingRight": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "yellowgreen",
            "fontFamily": "Avenir",
            "fontSize": 22,
            "fontWeight": "bold",
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select Metal Category for Recycle Instructions
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Aerosol Cans
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Aluminum Cans
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Aluminum Foil
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Aluminium Takeout Containers
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Metal Caps and Lids
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Scrap Metal
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Tin or Steel Cans
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
